bin/backend_check

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+"""
+This script checks that all terraform directories under the directory provided contain backend configs.
+
+Usage:
+    backend_check [options] [PATH]
+
+Arguments:
+    PATH    The path of the directory to check. Defaults to the current working directory.
+
+Options:
+    -h, --help          Show this message and exit.
+    --version           Show the version.
+"""
+import os
+import sys
+
+import hcl
+from docopt import docopt
+
+from terrawrap.version import __version__
+
+
+def main():
+    arguments = docopt(__doc__, version=__version__)
+    target_dir = arguments["PATH"] if arguments["PATH"] else os.getcwd()
+
+    config_dirs = {
+        current_dir: [file for file in files if file.endswith(".tf")]
+        for current_dir, dirs, files in os.walk(target_dir, followlinks=True)
+        if ".terraform" not in current_dir
+           and any(entry.endswith(".tf") for entry in files)
+    }
+
+    config_dirs_with_terraform_backend = set()
+
+    for config_dir, files in config_dirs.items():
+        # Traditionally we put this in a backend.tf file, so check for it there first.
+        if "backend.tf" in files:
```

Comment:
Do we really need this? I'm not sure if the extra few ms this saves us is worth the extra complexity in the code